project_template/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.15)
+
+project(az_exampleshortname
+    LANGUAGES C CXX
+    VERSION 0.1.0)
+
+option(BUILD_DOCS "Build documentation" ON)
+option(BUILD_TESTING "Build tests" ON)
+set(AZ_INSTALL_CMAKEPACKAGEDIR "share/cmake" CACHE STRING "directory to install cmake packages into")
+
+# if needed: add ./cmake to the module path
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 17)

```

Comment:
Usually the enforced standard is the minimal version required for the code. C99 seems fine, just wondering why do we require C++ 17? This would exclude cases where customers want to use a lower version - assuming our code does not really require C++17 of course.

Reply:
that's not what CMAKE_CXX_STANDARD does unless you also set CMAKE_CXX_STANDARD_REQUIRED (same for C, but with C instead of CXX).

Reply:
Got it. Looked closer at these - learn something new everyday.

---
In reply to: [309450713](https://github.com/Azure/azure-sdk-for-c/pull/2#discussion_r309450713) [](ancestors = 309450713)